File: leetcode problem/DSA_Surya_Sir_Questions/Day2.cpp
#include <vector>

// 169. Majority Element

class Solution {
public:
    int majorityElement(std::vector<int>& nums) {
        int count = 0;
        int candidate = 0;

        for (int num : nums) {
            if (count == 0) {
                candidate = num;
            }

            if (num == candidate)
                count++;
            else
                count--;
        }

        return candidate;
    }
};


// 26. Remove Duplicates from Sorted Array

class Solution {
public:
    int removeDuplicates(std::vector<int>& nums) {
        if (nums.size() == 0) {
            return 0;
        }

        int k = 1;

        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] != nums[k - 1]) {
                nums[k] = nums[i];
                k++;
            }
        }

        return k;
    }
};

File: day_2.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = 0
        m = 0

        for i in nums:
            if count == 0:
                m = i

            if i == m:
                count += 1
            
            else:
                count -= 1
            
        return m
    
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        currsum = nums[0]
        maxsum = nums[0]
        for i in range(1 , len(nums)):
            currsum = max(nums[i] , currsum + nums[i])
            maxsum = max(currsum , maxsum)

        return maxsum

class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        for i in nums:
            if i == 0:
                nums.remove(0)
                nums.append(0)
            else:
                continue
                return nums    